File: python/isbn-verifier/isbn_verifier.py
def verify(isbn):
    s = isbn.replace('-', '')

    return len(s) == 10 and all(map(str.isdigit, s[:9])) and (s[9] == 'X' or s[9].isdigit()) and sum(map(lambda i: parse_int(s[i]) * (10 - i), range(10))) % 11 == 0


def parse_int(ch):
    return 10 if ch == 'X' else (ord(ch) - ord('0'))

File: haskell/isbn-verifier/src/IsbnVerifier.hs
module IsbnVerifier (isbn) where

import Data.Char
import Data.String.Utils

isbn :: String -> Bool
isbn str = let s = replace "-" "" str
           in length s == 10
            && all isDigit (init s)
            && (s !! 9) `elem` (['0'..'9'] ++ "X")
            && (sum [(parseInt (s !! i)) * (10 - i) | i <- [0..9]]) `mod` 11 == 0

parseInt :: Char -> Int
parseInt ch = if ch == 'X' then 10 else ord(ch) - ord('0')